specs/eip4844/validator.md

```diff
@@ -0,0 +1,135 @@
+# EIP-4844 -- Honest Validator
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Helpers](#helpers)
+  - [`is_data_available`](#is_data_available)
+  - [`verify_blobs_sidecar`](#verify_blobs_sidecar)
+- [Beacon chain responsibilities](#beacon-chain-responsibilities)
+  - [Block proposal](#block-proposal)
+    - [Constructing the `BeaconBlockBody`](#constructing-the-beaconblockbody)
+      - [Blob commitments](#blob-commitments)
+  - [Beacon Block publishing time](#beacon-block-publishing-time)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document represents the changes to be made in the code of an "honest validator" to implement EIP-4844.
+
+## Prerequisites
+
+This document is an extension of the [Bellatrix -- Honest Validator](../altair/validator.md) guide.
+All behaviors and definitions defined in this document, and documents it extends, carry over unless explicitly noted or overridden.
+
+All terminology, constants, functions, and protocol mechanics defined in the updated [Beacon Chain doc of EIP4844](./beacon-chain.md) are requisite for this document and used throughout.
+Please see related Beacon Chain doc before continuing and use them as a reference throughout.
+
+## Helpers
+
+### `is_data_available`
+
+The implementation of `is_data_available` is meant to change with later sharding upgrades.
+Initially, it requires every verifying actor to retrieve the matching `BlobsSidecar`,
+and verify the sidecar with `verify_blobs`.
+
+Without the sidecar the block may be processed further optimistically,
+but MUST NOT be considered valid until a valid `BlobsSidecar` has been downloaded.
+
+```python
+def is_data_available(slot: Slot, beacon_block_root: Root, kzgs: Sequence[KZGCommitment]):
+    sidecar = retrieve_blobs_sidecar(slot, beacon_block_root)  # implementation dependent, raises an exception if not available
+    verify_blobs_sidecar(slot, beacon_block_root, kzgs, sidecar)
+```
+
+### `verify_blobs_sidecar`
+
+```python
+def verify_blobs_sidecar(slot: Slot, beacon_block_root: Root,
+                         expected_kzgs: Sequence[KZGCommitment], blobs_sidecar: BlobsSidecar):
+    assert blobs_sidecar.shard == 0  # always zero, placeholder for future sharding
+    assert slot == blobs_sidecar.beacon_block_slot
+    assert beacon_block_root == blobs_sidecar.beacon_block_root
```

Comment:
Should we verify that `beacon_block.slot == beacon_block_slot`? You can only have a sidecar for non-empty slots